src/Entity/MetadataDisplayEntity.php

```diff
@@ -473,95 +507,57 @@ public function getTwigVariablesUsed() {
-   *
    * @return array
    *   A list of used $variables by this template.
    */
-  private function getTwigVariableNames(ModuleNode|Node|BodyNode $nodes, array $all_variables, string $set_var = '', string $set_source = ''): array {
+  private function getTwigVariableNames(ModuleNode|Node|BodyNode $nodes): array {
     $variables = [];
     foreach ($nodes as $node) {
       $lineno = [$node->getTemplateLine()];
-      $variable_key = '';
-      $parent_path = '';
-      if ($node instanceof NameExpression && !$nodes instanceof NotUnary) {
-        if (!$node->hasAttribute('name')) {
-          continue;
-        }
-        $variable_key = $node->getAttribute('name');
-        if ($variable_key == '_key') {
+      $variable_key = $node->hasAttribute('name')
+                      && $node->getAttribute('name') == 'data'
+        ? $node->getAttribute('name')
+        : '';
+      if ($node->hasAttribute('always_defined')) {
+        if ($node->getAttribute('always_defined')
+            && $nodes->hasNode('seq')) {
           $seq = $nodes->getNode('seq');
-          if ($seq->hasNode('node') && $seq->getNode('node')->hasAttribute('name')) {
+          if ($seq->hasNode('node')
+              && $seq->getNode('node')->hasAttribute('name')
+              && $seq->getNode('node')->getAttribute('name') == 'data'
+          ) {
             $seq_name = $seq->getNode('node')->getAttribute('name');
           }
-          elseif ($seq->hasAttribute('name')) {
+          elseif ($seq->hasAttribute('name')
+                  && $seq->getAttribute('name') == 'data') {
             $seq_name = $seq->getAttribute('name');
           }
-          else {
-            continue;
+          if (!empty($seq_name)) {
+            $seq_value = $seq->hasNode('attribute')
+              ? $seq->getNode('attribute')->getAttribute('value')
+              : '';
+            $variable_key = empty($seq_value) ? $seq_name : $seq_name . '.' . $seq_value;
           }
-          $seq_value = $seq->hasNode('attribute') ? $seq->getNode('attribute')->getAttribute('value') : '';
-          $value_target = $nodes->getNode('value_target');
-          if (!$value_target->hasAttribute('name')) {
-            continue;
-          }
-          $variable_key = $value_target->getAttribute('name');
-          $parent_path = empty($seq_value) ? $seq_name : $seq_name . '.' . $seq_value;
-          $set_var = $variable_key;
-          $set_source = $parent_path;
         }
       }
-      elseif ($node instanceof ConstantExpression && $nodes instanceof GetAttrExpression) {
-        $variable_key = $node->getAttribute('value');
+      elseif ($node->hasAttribute('value')
+              && $nodes->hasNode('node')
+              && $nodes->getNode('node')->hasAttribute('name')
+              && $nodes->getNode('node')->getAttribute('name') == 'data'
+      ) {
+        $variable_key = $nodes->getNode('node')->getAttribute('name')
```

Comment:
Everywhere here $variable_key (first part) will always start with `data` (because of the conditions). I see how you could save  a lot of IFs by grouping conditionals and also setting some values as fixed strings (e.g data)